pkg/signals/signal.go

```diff
@@ -0,0 +1,31 @@
+package signals
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+var (
+	onlyOneSignalHandler = make(chan struct{})
+	shutdownSignals = []os.Signal{os.Interrupt, syscall.SIGTERM}
+)
+
+// SetupSignalHandler registers for SIGTERM and SIGINT. A stop channel is returned
+// which is closed on one of these signals. If a second signal is caught, the program
+// is terminated with exit code 1.
+func SetupSignalHandler() (stopCh chan struct{}) {
```

Comment:
Created a [similar](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/manager/signals/signal.go) function locally that returns a stop channel that can send and receive.